# BIOE183

## Course notes:
1)
2)
3)
4)
5)

## Raw Sample files:
We will use a dataset from a previous [EBI traning courses](https://www.ebi.ac.uk/training/online/course/ebi-next-generation-sequencing-practical-course/rna-sequencing/rna-seq-analysis-transcriptome), the data is derived from sequencing of mRNA from zebrafish embryos in two diï¬€erent developmental stages. Sequencing was performed on the Illumina platform
and generated 76bp paired-end sequence data using poly-(A)+ selected RNA. 
- [2cells_1.fastq](ftp://ftp.ebi.ac.uk/pub/training/Train_online/RNA-seq_exercise/2cells_1.fastq) 
- [2cells_2.fastq](ftp://ftp.ebi.ac.uk/pub/training/Train_online/RNA-seq_exercise/2cells_2.fastq)
- [6h_post_fertilisation_R1.fastq](ftp://ftp.ebi.ac.uk/pub/training/Train_online/RNA-seq_exercise/6h_1.fastq)
- [6h_post_fertilisation_R2.fastq](ftp://ftp.ebi.ac.uk/pub/training/Train_online/RNA-seq_exercise/6h_2.fastq)

## Cleaning Fastq
- [2cells_R1_clean.fastq](https://sysbio.ucsd.edu/public/wenxingzhao/CourseFall2019/cleanReads/2cells_R1_clean.fastq)
- [2cells_R2_clean.fastq](https://sysbio.ucsd.edu/public/wenxingzhao/CourseFall2019/cleanReads/2cells_R1_clean.fastq)
- [6h_R1_clean.fastq](https://sysbio.ucsd.edu/public/wenxingzhao/CourseFall2019/cleanReads/6h_R1_clean.fastq)
- [6h_R2_clean.fastq](https://sysbio.ucsd.edu/public/wenxingzhao/CourseFall2019/cleanReads/6h_R2_clean.fastq)
## Sample alignment files 

## Sample read count table

## R Script

## Recourses:
#### 1.Tutorials:
[1] Weill Cornell Medical Colledge: http://chagall.med.cornell.edu/RNASEQcourse/

#### 2.Software manuals:
- [Bioconda](https://www.youtube.com/watch?v=lGa9PCSH5IU) starting from 3:30. 
- [fastQC manual](https://dnacore.missouri.edu/PDF/FastQC_Manual.pdf)
- [fastp manual](https://github.com/OpenGene/fastp)

